socket.io.js

```diff
@@ -49,20 +49,37 @@ function lookup(uri, opts) {
   var parsed = url(uri);
   var source = parsed.source;
   var id = parsed.id;
+  var path = parsed.path;
+  var host = parsed.host;
+  var port = parsed.port;
+  var scheme = parsed.protocol;
   var io;
 
   if (opts.forceNew || opts['force new connection'] || false === opts.multiplex) {
     debug('ignoring socket cache for %s', source);
     io = Manager(source, opts);
-  } else {
-    if (!cache[id]) {
-      debug('new io instance for %s', source);
-      cache[id] = Manager(source, opts);
+    return io.socket(path);
+  }
+
+  if (cache[id]) {
+    var cachedPath = cache[id].nsps[path].nsp;
+    var cachedHost = cache[id].opts.host;
+    var cachedPort = cache[id].opts.port;
+    var cachedScheme = cache[id].uri.substr(0, cache[id].uri.indexOf(':'));
```

Comment:
If the `id` matched, it means everything else matched. Look at how `id` is constructed first. I think you might only need to compare `.path` ?